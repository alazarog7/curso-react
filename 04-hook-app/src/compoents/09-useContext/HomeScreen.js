import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const HomeScreen = () => {
    
    const context = useContext(UserContext);
    console.log(context);
    return (
        <>
            <h1>HomeScreen</h1>
            <hr/>

            <pre>{JSON.stringify(context, null, 3)}</pre>
        </>
    )
}
