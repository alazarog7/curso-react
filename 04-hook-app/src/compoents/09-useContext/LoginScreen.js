import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const context = useContext(UserContext);

    console.log(context);
    return (
        <>
            <h1>LoginScreen</h1>
            <hr/>

            <button className="btn btn-outline-primary" onClick={
                ()=>{
                    context.setUser({
                        id:1234,
                        name:'Alejandro Lazaro Gutierrez'
                    })
                }
            }> Login </button>
        </>
    )
}
