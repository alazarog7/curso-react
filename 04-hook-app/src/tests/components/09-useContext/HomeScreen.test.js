import { mount, shallow } from "enzyme"
import { HomeScreen } from "../../../compoents/09-useContext/HomeScreen"
import { UserContext } from "../../../compoents/09-useContext/UserContext";

describe('Tests HomeScreen', () => {
    const user = {
        name: 'Alejandro',
        email: 'a.lazaro.g7@gmail.com'
    }
    
    test('should show the component ', () => {
        const wrapper = mount(
            <UserContext.Provider value={{
                user
            }}>
                <HomeScreen />
            </UserContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
    })
    
})
