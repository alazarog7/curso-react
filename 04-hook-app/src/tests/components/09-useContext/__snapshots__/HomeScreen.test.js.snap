// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests HomeScreen should show the component  1`] = `
Array [
  <h1>
    HomeScreen
  </h1>,
  <hr />,
  <pre>
    {
   "user": {
      "name": "Alejandro",
      "email": "a.lazaro.g7@gmail.com"
   }
}
  </pre>,
]
`;
